rgdps/common/validators.py

```diff
@@ -28,3 +35,59 @@ def _validate(cls, value: Any, _: core_schema.ValidationInfo) -> Base64String:
             return Base64String(hashes.decode_base64(value))
         except Exception as e:
             raise ValueError(f"Input is not valid base64") from e
+
+
+class TextBoxString(str):
+    @classmethod
+    def __get_pydantic_core_schema__(
+        cls,
+        _: type[Any],
+    ) -> core_schema.CoreSchema:
+        return core_schema.general_after_validator_function(
+            cls._validate,
+            core_schema.str_schema(),
+        )
+
+    @classmethod
+    def _validate(cls, value: Any, _: core_schema.ValidationInfo) -> TextBoxString:
+        if not isinstance(value, (str, bytes)):
+            raise TypeError("Value must be str or bytes")
+
+        if isinstance(value, bytes):
+            value = value.decode()
+
+        # Value needs to be: stripped and alphanumeric.
+        value = value.strip()
+
+        if not re.match(r"^[a-zA-Z0-9 ]+$", value):
```

Comment:
Would pre-compining the regex be beneficial?

Reply:
i think you can just use https://docs.python.org/3/library/stdtypes.html#str.isalnum
```suggestion
        if not value.isalnum():
```

Reply:
oo actually no because of the spaces

Reply:
Was `str.isalnum` prior, addressed in previous review.